### ISSUE TITLE
False positive in `RedundantSuspendModifierSpec` when calling suspending method from test

### ISSUE THREAD

**Issue by kylannjohnson** [detekt/detekt#7757]
## Expected Behavior
I have a test that wraps a set of assertions into a suspending method to reduce duplication.  Of those assertions is a call to a suspending method so the assertion method is also labeled with `suspend`.  I would expect this to pass detekt.

## Observed Behavior
A `RedundantSuspendModifier` issue is raised in the above scenario.  When placed in `RedundantSuspendModifierSpec`, the following test will fail, but I would expect it to pass. 

```kotlin
    @Test
    fun `no reports when suspending method is called from test`() {
        val code = """
            import kotlinx.coroutines.CoroutineScope
            import kotlinx.coroutines.Dispatchers
            import kotlinx.coroutines.launch
            import kotlinx.coroutines.coroutineScope
            import kotlinx.coroutines.test.runTest

            class A {
                annotation class Test

                @Test
                fun `test that calls suspending method`() = runTest {
                    assertion()
                }
                
                suspend fun assertion() {
                     delay(1_000)  // emulates calling a suspending method on a test fake
                }
            }
        """.trimIndent()
        assertThat(subject.compileAndLintWithContext(env, code)).isEmpty()
    }
```

if you wrap the body of the `assertion()` method above with `coroutineScope`, the test will pass.  This could be the suggested fix for the issue, but I wouldn't call that a "redundant suspend modifier"

## Context
I had a group of assertions that were duplicated across tests and I wanted to consolidate them into one method.  One of the assertions interacts with a Flow, so it is suspending so i made the new method suspending.

## Your Environment
* Version of detekt used: 1.23.7
* Version of Gradle used (if applicable): 8.10.2

**Comment by kylannjohnson** [comment]
Hey @atulgpt .  That does make the test pass, but my issue still remains.   If I don't include `coroutineScope` on the `assertion()` method, I get a `RedundantSuspendModifier` issue raised for that method.   Do you know of any insight or possible ways to replicate this case?

**Comment by atulgpt** [comment]
Hi @kylannjohnson can you add the updated failing TC(with the `delay` import) as I am not getting ` If I don't include coroutineScope on the assertion() method, I get a RedundantSuspendModifier issue raised for that method.`?

**Comment by kylannjohnson** [comment]
Hey @atulgpt, sorry for the confusion.  When I added the import of `delay`, the test did pass, as you said.  The test is unsuccessful in replicating my issue.   The suspending method I added to my test is the one that raises the `RedundantSuspendModifier` issue, but I can't seem to craft a test to show it.   However, the issue still exists on my side.  If i add `coroutineScope` to the method, no issue is raised.   

The test below is closer in structure to the code on my side

```kotlin
    @Test
    fun `no reports when suspending method is called from test`() {
        val code = """
            import kotlinx.coroutines.CoroutineScope
            import kotlinx.coroutines.Dispatchers
            import kotlinx.coroutines.ExperimentalCoroutinesApi
            import kotlinx.coroutines.delay
            import kotlinx.coroutines.launch
            import kotlinx.coroutines.coroutineScope
            import kotlinx.coroutines.test.runTest

            @OptIn(ExperimentalCoroutinesApi::class)
            class SuspendingType {
               suspend fun suspendSomething() {
                   delay(1_000) // Only here to not make this method fail the check
               }
            }

            val suspendingType : SuspendingType 
                get() = SuspendingType()
            
            class A {
                annotation class Test

                @Test
                fun `test that calls suspending method`() = runTest {
                    assertion()
                }
                
               // In my code, wrap this method's body in `coroutineScope` to pass the issue.
                private suspend fun assertion() {
                     suspendingType.suspendSomething()  // emulates calling a suspending method on a test fake
                }
            }
        """.trimIndent()
        assertThat(subject.compileAndLintWithContext(env, code)).isEmpty()
    }
```

Again, on my side, I would add `coroutineScope` to the definition of the `assertion()` method above to get it to pass.   I'll have to research a bit on how Detekt treats this case to understand how to replicate the issue.  If you have any suggestions, I can research them too.

**Comment by atulgpt** [comment]
Hi, @kylannjohnson just to be sure, you're running detekt with type resolution. Right? Also in your TC is there an unrecognised dependency or method that could throw the analysis away and give false results?
Also is this possible to create a sample project replicating your issue?

**Comment by BraisGabin** [comment]
Closing as not reproducible.
